### ISSUE TITLE
Profiler: Associate point tasks with their producing index launch

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#1347]
For example, we would like to see on each point task box when it was launched (or even mark task launches on the launching task's timeline). We don't care about the creation time of the point task itself, but the original index launch. Also, the runtime work that handles an index launch as a whole (e.g. the logical dependence analysis) is not currently connected with the point tasks themselves, or the runtime work handling each point task in isolation (e.g. the physical dependence analysis).

The timestamps we get from Realm only cover the point tasks in isolation, so the index launch information would need to be propagated in some way inside Legion.

Note that this propagation might be multi-stage, since an index task can be copied multiple times, and each copy gets a different unique id, e.g. in the case of recursive slicing, or remote mapping.

CC @lightsighter @eddy16112 @seemamirch

**Comment by elliottslaughter** [comment]
I think @lightsighter did this with the `Creator` item link?

**Comment by lightsighter** [comment]
No, we haven't made all of those connections yet. For example, there is still no path for associating the physical analysis for a point task with the logical analysis of its corresponding index space task.
